plugins/jobtype/src/azkaban/jobtype/HadoopConfigurationInjector.java

```diff
@@ -50,6 +52,17 @@
    * configuration properties to automatically inject.
    */
   public static void injectResources() {
+    // Add mapred, yarn and hdfs site configs (in addition to core-site, which
+    // is automatically added) as default resources before we add the injected
+    // configuration. This will cause the injected properties to override the
+    // default site properties (instead of vice-versa). This is safe to do,
+    // even when these site files don't exist for your Hadoop installation.
+    Configuration.addDefaultResource("mapred-default.xml");
```

Comment:
Since we don't exactly know what side effects this will have, can you add a guard around these 4 lines of code.  This way we can turn off this feature if needed via a global property.

Something like this.

if (props.getBoolean("azkaban.inject.hadoop-site.configs", true)) {
}